public/app.js

```diff
@@ -216,5 +217,36 @@
         });
 
         $httpProvider.interceptors.push('jwtInterceptor');
-    }
+
+        $translateProvider
+            .useSanitizeValueStrategy('sanitizeParameters')
+            .useStaticFilesLoader({
+                prefix: 'static/languages-<%= VERSION %>/',
+                suffix: '.json'
+            })
+            .determinePreferredLanguage(function () {
+                let lang = 'en'; // default language
+
+                if (navigator.language) {
+                    lang = navigator.language;
+                }
+
+                // if it is set via query, use that
+                const queries = document.location.search.substr(1).split('&');
+                for (query in queries) {
+                    if (query.startsWith('lang=')) {
+                        lang = query.substr('lang='.length);
+                        break;
+                    }
+                }
+
+                // shorten en-XX to en
+                if (lang.startsWith('en')) {
```

Comment:
Speaking of weird browser behaviour, apparently [old Safari versions would return this string lower-cased](https://developer.mozilla.org/en-US/docs/Web/API/NavigatorLanguage/language)

About 25% of Safari users are using versions of Safari which would be affected by this. 

So I'm tempted to say we should do `lang = lang.toLowerCase()` and just make everything lower-case for consistency across the board.

Reply:
👍

Reply:
To be fair though.... it's 25% of Safari users. And Safari users are 10% overall. So it's a small number we're talking about. 

But I'm in edge-case-hunting-mode tonight.  :)